Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -849,9 +850,13 @@ static LayoutUnit computeGridSpanSize(const Vector<GridTrack>& tracks, const Gri
 
 std::optional<LayoutUnit> GridTrackSizingAlgorithm::gridAreaBreadthForGridItem(const RenderBox& gridItem, GridTrackSizingDirection direction) const
 {
-    if (m_renderGrid->areMasonryColumns())
+    if (m_renderGrid->areMasonryColumns() && direction == GridTrackSizingDirection::ForColumns)
         return m_renderGrid->contentLogicalWidth();
 
+
+    if (m_renderGrid->areMasonryRows() && direction == GridTrackSizingDirection::ForRows && !GridLayoutFunctions::isOrthogonalGridItem(*m_renderGrid, gridItem))
+        return m_renderGrid->contentLogicalHeight();
+
```

Comment:
While I think this is ok for now (in order to fix these tests), I think this code only makes sense if we have already computed the logical width/height of the grid. If we are currently computing the logical width of the grid (e.g. when we are computing the intrinsic widths of the grid) or before we have done masonry layout/placement, then this is probably does not return the right value. If you plan to fix this in a follow up patch then I think you should at least add a `FIXME` here.